modules/bitgo/test/v2/unit/tss/ecdsa.ts

```diff
@@ -382,4 +391,140 @@ describe('Ecdsa tss helper functions tests', function () {
     });
   });
 
+  describe('signing share parsers', function() {
+    let userSignShare: ECDSAMethodTypes.SignShare;
+    let bitgoSignShare: ECDSA.SignConvertRT;
+    let userGammaAndMuShare: ECDSAMethodTypes.CreateUserGammaAndMuShareRT;
+    let userOmicronAndDeltaShare: ECDSAMethodTypes.CreateUserOmicronAndDeltaShareRT;
+    let bitgoOmicronAndDeltaShare: ECDSAMethodTypes.CreateUserOmicronAndDeltaShareRT;
+    let userSignatureShare: ECDSAMethodTypes.SignatureShare;
+    before(async function() {
+      const txRequest = {
+        txRequestId: 'randomId',
+        unsignedTxs: [{ signableHex: 'TOO MANY SECRETS', serializedTxHex: 'randomhex2' }],
+        signatureShares: [
+          {
+            from: 'bitgo',
+            to: 'user',
+            share: '',
+          }],
+      };
+      const signablePayload = Buffer.from(txRequest.unsignedTxs[0].signableHex, 'hex');
+
+      userSignShare = await ECDSAMethods.createUserSignShare(userKey.xShare, userKey.yShares[3]);
+      bitgoSignShare = mpc.signConvert({
+        kShare: userSignShare.kShare,
+        xShare: bitgoKey.xShare,
+        yShare: bitgoKey.yShares[1],
+      });
+
+      userGammaAndMuShare = await ECDSAMethods.createUserGammaAndMuShare(userSignShare.wShare, bitgoSignShare!.aShare!);
+      const bitgoGammaShare = mpc.signConvert({
+        bShare: bitgoSignShare.bShare,
+        muShare: userGammaAndMuShare.muShare,
+      });
+
+      bitgoOmicronAndDeltaShare = mpc.signCombine({
+        gShare: bitgoGammaShare.gShare!,
+        signIndex: {
+          i: 1,
+          j: 3,
+        },
+      });
+      userOmicronAndDeltaShare = await ECDSAMethods.createUserOmicronAndDeltaShare(userGammaAndMuShare.gShare!);
+
+      userSignatureShare = await ECDSAMethods.createUserSignatureShare(userOmicronAndDeltaShare.oShare, bitgoOmicronAndDeltaShare.dShare, signablePayload);
+    });
+
+    it('should successfully parse K share', function () {
+      const bitgoKShare = userSignShare.kShare;
```

Comment:
These tests looks nice, Probably we can add a addition check on the parsers to make sure the share has sufficient length and to throw an error if it doesn't meet the condition also